operator/pkg/central/extensions/reconcile_tls.go

```diff
@@ -261,7 +260,7 @@ func (r *createCentralTLSExtensionRun) reconcileScannerV4DBTLSSecret(ctx context
 }
 
 func (r *createCentralTLSExtensionRun) validateServiceTLSData(serviceType storage.ServiceType, fileNamePrefix string, fileMap types.SecretDataMap) error {
-	if err := certgen.VerifyServiceCert(fileMap, r.ca, serviceType, fileNamePrefix); err != nil {
+	if err := certgen.VerifyCert(fileMap, fileNamePrefix, r.getValidateCert(serviceType)); err != nil {
```

Comment:
`certgen.VerifyServiceCert` is not used anymore and the only usage of `certgen.VerifyCert` is here in `reconcile_tls.go`.

Maybe we could:
1) Remove `certgen.VerifyServiceCert`
2) simplify `certgen.VerifyCert` and inline `r.getValidateCert(serviceType)` (with `certgen.GetValidateServiceCertFunc(r.ca, serviceType)` function there?

With a single usage, the function wrap now looks excessive.

Reply:
Good point. Filed https://issues.redhat.com/browse/ROX-21762 to track the cleanups.